flocker/node/_deploy.py

```diff
@@ -817,6 +830,32 @@ def discover_state(self, local_state):
                 manifestations=None,
                 paths=None,
             )])
+        # Declare ignorance if any of the reported manifestations are found not
+        # mounted (manifest).
+        # This may occur after a reboot, if the container agent connects to the
+        # control service before the dataset agent and receives the (now stale)
+        # dataset state reported prior to the reboot.
+        # In that situation a stateful container may be started before the
+        # dataset agent has attached and mounted its dataset.
+        all_manifest = True
+        for manifestation in local_state.manifestations.values():
+            if not _is_mounted(
+                    # XXX This isn't right. Need a clean way to get the dataset
+                    # mountpoint from here. How?
+                    u'/flocker/{}'.format(manifestation.dataset.dataset_id)
```

Comment:
That's definitely wrong (e.g. ZFS backend won't work with this since it uses different format). We need to operate on the path which is definitely going to be passed to docker -- that was possible with the previous implementation approach...